pytorch_lightning/trainer/distrib_data_parallel.py

```diff
@@ -411,13 +413,15 @@ def set_random_port(self, force=False):
         """
         # pick a random port first
         assert self.num_nodes == 1, 'random port can only be called from single node training'
-        global RANDOM_PORTS
-        default_port = RANDOM_PORTS[-1]
-        RANDOM_PORTS = RANDOM_PORTS[:-1]
+
+        print('setting port on rank', self.global_rank)
+        default_port = os.environ.get('MASTER_PORT')
```

Comment:
Not sure if relevant...but when I pulled this current version of PR down and ran gpu_template.py, I got an NCCL error. In an[ issue I found on github](https://github.com/pytorch/pytorch/issues/20313), someone said this was because "master_port was used more than once at the same time". 

How are we making sure this isn't happening? (or how can we?)

Reply:
I don't know what this person means in the post. This is the port that the processes use to communicate through. It is expected that it gets used multiple times.